src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -366,9 +376,9 @@ internal void OnIdle(object state)
         }
 
         // Internal for testing
-        internal void ReparseOnForeground(object state)
+        internal void ReparseOnUIThread(object state)
         {
-            _dispatcher.AssertForegroundThread();
+           _joinableTaskContext.AssertUIThread();
```

Comment:
Judging by the call sites in this file it looks like you're dispatching to this method via the _dispatcher. For what it's worth I actually don't think this class needs the _dispatcher at all. Might make this easier.